lms/validation/_bearer_token.py

```diff
@@ -109,14 +107,10 @@ def lti_user(self):
         :rtype: LTIUser
         """
         try:
-            return parser.parse(
-                self,
-                self.context["request"],
-                locations=["headers", "querystring", "form"],
-            )
-        except HTTPUnprocessableEntity as error:
+            return self.parse(locations=["headers", "querystring", "form"])
+        except ValidationError as error:
             try:
-                authorization_error_message = error.json["authorization"][0]
+                authorization_error_message = error.messages["authorization"][0]
```

Comment:
This change is a little tricky to understand. There's a `_handle_error()` hook method below that causes validation to raise `lms.validation.ValidationError` instead of the webargs Pyramid integration's default behaviour of raising a Pyramid `HTTPUnprocessableEntity`. Previously `_bearer_token.py` and `_lti_launch.py` imported `webargs.pyramidparser.parser` themselves and used _their own_ copy of `parser` to do the parsing. Our `_handle_error` hook wasn't attached to that parser so it raised the default `HTTPUnprocessableEntity` and we had to catch and wrap that here. Now they call `self.parse()` instead so they do use our one, customised copy of the parser (the one from `PyramidRequestSchema`), so `parse()` raises `ValidationError` not `HTTPUnprocessableEntity`.